pkg/handler/eval_cache.go

```diff
@@ -70,13 +92,43 @@ func (ec *EvalCache) GetByTags(tags []string) []*entity.Flag {
 			}
 		}
 	}
+	return results
+}
 
-	values := make([]*entity.Flag, 0, len(results))
-	for _, f := range results {
-		values = append(values, f)
+func (ec *EvalCache) getByTagsALL(tags []string) map[uint]*entity.Flag {
+	results := map[uint]*entity.Flag{}
+	cache := ec.cache.Load().(*cacheContainer)
+
+	for i, t := range tags {
+		fSet, ok := cache.tagCache[t]
+		if !ok {
+			// clear as no flags
+			results = map[uint]*entity.Flag{}
+			continue
+		}
+
+		if i == 0 {
+			// store all the flags
+			for fID, f := range fSet {
+				results[fID] = f
+			}
+		} else {
+			if len(results) > 0 {
+				// accept only previous flags
+				missingFIDs := []uint{}
+				for fID := range results {
+					if fSet[fID] == nil {
+						missingFIDs = append(missingFIDs, fID)
+					}
+				}
+				for _, fID := range missingFIDs {
+					delete(results, fID)
+				}
```

Comment:
How about one pass of the keys in results, and just delete the keys not the current fSet?

```
				for fID := range results {
					if _, ok := fSet[fID]; !ok {
						delete(results, fID)
					}
				}
```

Reply:
Cool, I wasn't sure if in Go is possible to modify a slice while iterating it. Changing it for sure

Reply:
related topic https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-map-within-a-range-loop, I also learned that `delete` marks the keys as deleted, thus safe to keep iterating.